lib/authz/permissions.go

```diff
@@ -510,6 +532,10 @@ func (a *authorizer) authorizeAdminAction(ctx context.Context, authContext *Cont
 		return trace.Errorf("failed to validate MFA auth response, authorizer missing mfaAuthenticator field")
 	}
 
+	if mfaResp.GetWebauthn() == nil {
+		return trace.AccessDenied("admin actions require webauthn device verification to complete")
+	}
```

Comment:
Is this an optimization because we expect the `ValidateMFAAuthResponse` to fail, or a change in behavior?

Reply:
This is to prevent TOTP from being used for Admin MFA, since we are opening up to `second_factor: optional`.